Automattic/pocket-casts-ios
# Transcripts: handle diacritics when searching

| 📘 Part of: #1848 | 
|:---:|

Fixes #2039

Handles diacritics when searching.

<img src="https://github.com/user-attachments/assets/2814c117-7549-4b01-93f3-dc55a4187be8" width="300">

I also improved `KMPSearch` to cache the transcript content, avoiding parsing the transcript every time a new search term is entered.

## To test

1. Ensure **transcripts** feature flag is enabled
2. Run the app
3. Play any Chaosradio episode
4. Search for any word with accents or special characters without adding those
5. ✅ The results should match
6. Add a new episode with transcript to your Up Next
7. Move the scrubber to the end of the current episode
8. When the new one starts playing, try searching on it
9. ✅ Search should match correctly

## Checklist
